test/Microsoft.AspNetCore.JsonPatch.Test/JsonPropertyComplexNameDTO.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Newtonsoft.Json;
+
+namespace Microsoft.AspNetCore.JsonPatch
+{
+    public class JsonPropertyComplexNameDTO
+    {
+        [JsonProperty("foo/bar~")]
```

Comment:
It would be good to add some more cases that require escaping, such as:

Valid cases:

1. `foo/bar~`
2. `foo/bar~0baz`
3. `foo/bar~1baz`
4. `foo/bar~0/~0/~1~1/~0~0/baz`
5. `~0~1foo`

And perhaps more error cases as well:

1. `~`
2. `foo~`
3. `foo~2bar`